<launch>

  <arg name="bag_file" />
  <arg name="annotation_config_file" />

  <arg name="rviz" default="true" />

  <!-- play rosbag -->

  <param name="use_sim_time" value="true" />
  <node name="rosbag_play"
        pkg="rosbag" type="play"
        args="$(arg bag_file) --clock">
  </node>

  <!-- annotation for visualization -->

  <node name="transformable_interactive_server"
        pkg="jsk_interactive_marker" type="transformable_server_sample">
    <rosparam subst_value="true">
      display_interactive_manipulator: true
      display_interactive_manipulator_only_selected: true
    </rosparam>
  </node>
  <node name="transformable_boxes"
        pkg="fcn_3d" type="transformable_boxes_client"
        output="screen">
    <remap from="~server" to="transformable_interactive_server" />
    <rosparam subst_value="true">
      config_file: $(arg annotation_config_file)
      config_auto_save: true
    </rosparam>
  </node>

  <!-- TODO(wkentaro): Implement node to evaluate the segmentation -->

  <!-- visualization -->

  <group if="$(arg rviz)">
    <node name="$(anon rviz)"
          pkg="rviz" type="rviz"
          args="-d $(find icra2017)/rvizconfig/look_around_bin_rosbag.rviz">
    </node>
  </group>

</launch>
